### PR TITLE
configure --with-state-file: specify default state file

### PR DESCRIPTION
**Pull request by kdudka**

### CODE REVIEW

**Comment by glensc** [comment]
`--with-state-file-path` is more accurate in terms that `--without-state-file` may mean to disable state file feature.

File: configure.ac [hunk 1]
```diff
@@ -46,7 +46,26 @@ AS_CASE(["$with_acl"],
 AS_IF([test "$ac_cv_lib_acl_acl_get_file" = yes],
   echo "1" > ./test/test.ACL;, echo "0" > ./test/test.ACL;)
 
+AC_ARG_WITH([state-file],
+  AC_HELP_STRING([--with-state-file=PATH],
+                 [path to state file (/var/lib/lgrotate.status by default)]),
```
Comment (glensc) [inline configure.ac]: typo

**Comment by kdudka** [comment]
Thanks for review!  Resubmitted...

**Comment by glensc** [comment]
LGTM

**Pull request merged**